### ISSUE TITLE
PDF/UA-1 fails PAC (2026) checks on weird placement of raw code block figure

### ISSUE THREAD

**Issue by jassielof** [typst/typst#8703]
### Description

As reference the issue happens with (the thesis document) https://github.com/jassielof/upsa-bolivia-typst/tree/026053356a8ac04e194a13dc935061a198f11816.

The specific issue is:

<img width="1916" height="874" alt="Image" src="https://github.com/user-attachments/assets/76244cfe-8e8f-4593-ad1b-fd1269a791da" />

Which was generated with PAC on Windows. Weirdly because if I set such figure placement at the bottom, the issue disappears, if I prepend a page break to its section it also disappears, if I use have the following test file:

````typ
#set document(
  title: "TFG",
  author: "john",
)

// overall
#show figure: set figure.caption(position: top)
#show figure.where(kind: image): set block(breakable: false, sticky: true)
#show figure.where(kind: table): set block(breakable: true, sticky: false)
#show figure.where(kind: math.equation): set figure(supplement: [Fórmula])
#set figure(
  gap: 1.5em,
  placement: none,
)

// figure caption

#set figure.caption(separator: parbreak(), position: top)
#show figure.caption: set align(left)
#show figure.caption: set text(font: "Source Sans 3")
#show figure.caption: set par(first-line-indent: 0em)
#show figure.caption: it => {
  strong[#it.supplement #context it.counter.display(it.numbering)]
  parbreak()
  emph(it.body)
}

// raw only
#show figure.where(kind: raw): set figure(placement: none)
#show figure.where(kind: raw): set block(breakable: false, sticky: false, width: 100%)
#show figure.where(kind: raw): set raw(block: true)


= first heading

#lorem(10)

#figure(
  table(
    [hola],
    [adios],
  ),
  alt: "sample table",
)

#figure(
  ```c
  \documentclass{article}
  \begin{document}
    Hello, World!
  \end{document}
  ```,
  caption: [Sample TeX Code],
  alt: "tex code",
)

````

The issue doesn't happen.

The previous Discord discussion is on https://discord.com/channels/1054443721975922748/1054443722592497796/1532446501740216462.

### Operating system

Windows

### Typst version

- [x] I am using the latest version of Typst

**Comment by saecki** [comment]
The problem seems to be that we update the bounding box before we check if we're inside an artifact. For some reason the end tag of the figure tag is on the next page and the `ENTREGA` text on page 5 is considered to be inside the figure (which is kind of ok, because it's an artifact and thus should be ignored). The fix should be quite simple.

As a heads up, you shouldn't use `alt` if the figure contains a semantic element such as a table or a code block. The nature of an alternative description in PDF is that is must completely describe the content, because screen readers will only read the alternative description and not read any of the tags within such a figure. See: https://typst.app/docs/guides/accessibility/#:~:text=do%20not%20use%20the%20alt%20attribute%20of%20a%20figure%20if%20it%20contains%20a%20table%20element

**Comment by jassielof** [comment]
Thank you.
